queries/nim/injections.scm

```diff
@@ -0,0 +1,147 @@
+; =============================================================================
+; generalized_strings
+
+; regex in generalized_strings
+(generalized_string 
+  (identifier) @_string_prefix .
+  (string_literal "\"" . "\"") @injection.content
+  (#set! injection.language "regex")
+  (#offset! @injection.content 0 1 0 -1) ; wrapping `"` 
+  (#any-of? @_string_prefix "re" "rex"))
+
+; regex in multiline generalized_strings
+(generalized_string 
+  (identifier) @_string_prefix .
+  (string_literal "\"\"\"" . "\"\"\"") @injection.content
+  (#set! injection.language "regex")
+  (#offset! @injection.content 0 3 0 -3) ; wrapping `"""`
+  (#any-of? @_string_prefix "re" "rex"))
+
+; sql in generalized_strings
+(generalized_string 
+  (identifier) @_string_prefix .
+  (string_literal "\"" . "\"") @injection.content
+  (#set! injection.language "sql")
+  (#offset! @injection.content 0 1 0 -1) ; wrapping `"` 
+  (#eq? @_string_prefix "sql"))
+
+; sql in multiline generalized_strings
+(generalized_string 
+  (identifier) @_string_prefix .
+  (string_literal "\"\"\"" . "\"\"\"") @injection.content
+  (#set! injection.language "sql")
+  (#offset! @injection.content 0 3 0 -3) ; wrapping `"""`
+  (#eq? @_string_prefix "sql"))
+
+; there is no ts parser for peg
+; (generalized_string 
+;   (identifier) @_string_prefix .
+;   (string_literal "\"" . "\"") @injection.content
+;   (#set! injection.language "peg")
+;   (#offset! @injection.content 0 1 0 -1)
+;   (#eq? @_string_prefix "p[eE][gG]"))
+
+; format string in generalized_strings
+(generalized_string 
+  (identifier) @_string_prefix .
+  (string_literal) @injection.content
+  (#set! injection.language "nim_format_string")
+  (#eq? @_string_prefix "fmt"))
+
+; format string in normal strings with & prefix
+(prefix_expression
+  operator: (operator) @_string_prefix .
+  (string_literal) @injection.content
+  (#set! injection.language "nim_format_string")
+  (#eq? @_string_prefix "&"))
+
+; =============================================================================
+; asm statement
+
+; there is no ts parser for assembly
+; (assembly_statement (string_literal) @injection.content
+;   (#set! injection.language "assembly"))
+
+; =============================================================================
+; emit pragma
+
+; C / CPP / OBJC / JAVASCRIPT
+; a `#<space><language name>`
+; has to directly precede the {.emit: "<language code>".} pragma
+
+; eg.:
+; # objc
+; {.emit: "<objc code>".}
+; OR
+; # javascript
+; {.emit: "<javascript code>".}
+
+; normal strings
+((comment) @injection.language
+  (#offset! @injection.language 0 2 0 0) . ; leading `# `
+  (pragma_statement
+    (pragma_list
+      (colon_expression
+        left: (identifier) @_emit_keyword (#eq? @_emit_keyword "emit")
+        right: (string_literal "\"" . "\"") @injection.content)
+          (#offset! @injection.content 0 1 0 -1) ; wrapping `"`
+    )))
```

Comment:
push this offset to the end so that it's easier to read the predicate and query. Also, I think the parser should be doing the heavy lifting and create a string contents node here, and then it's captured for injection

```suggestion
((comment) @injection.language
 . 
 (pragma_statement
   (pragma_list
     (colon_expression
       left: (identifier) @_emit_keyword
       right: (string_literal "\"" . "\"") @injection.content)))
 (#eq? @_emit_keyword "emit")
 (#offset! @injection.language 0 2 0 0)
 (#offset! @injection.content 0 1 0 -1))

```

Reply:
@alaviss

Reply:
I can make this happen for strings, but only strings if that works for you.

Splitting comment would make this a non-terminal, which is not very nice in terms of `extras`.

Reply:
Only strings are more than enough.

I'm also okay with the current version, just that the code is a bit unreadable for me